### PR TITLE
fix: disable 'Accept' incoming funds buttons when other button was clicked

### LINKED ISSUE #3313: Individual "Accept" funds buttons should be disabled when "Accept all available" funds button is clicked and vice-versa
## Steps to reproduce

This is a pretty low priority issue.

* Step 1 - Switch to a metamask wallet
* Step 2 - Navigate to the incoming funds page
* Step 3 - Click the "Accept all available" funds button - do not accept the transaction

## Expected behaviour

All "Accept" funds buttons on individual claim rows should be disabled whilst the "Accept all available" funds transaction is processing.

The same should apply the opposite way round, if the "Accept" funds button on an individual claim is clicked, the "Accept all available" funds button should be disabled. All other "Accept" funds buttons should also be disabled as the "Accept" funds button claims *all* available incoming funds.

## Actual behaviour

The buttons are not disabled meaning it is possible to initiate multiple conflicting transactions getting the UI into a confused state.

https://github.com/user-attachments/assets/c5eef683-f9aa-40e6-a280-488f1296397f

### PR DESCRIPTION
**Pull request by Nortsova**
## Description

- [x] Disable "Accept" buttons when "Accept all available" is loading
- [x] Disable other "Accept" and "Accept all available" buttons when any "Accept" is loading

## Testing

* Step 1. Login with metamask
* Step 2. Clean Metamask history (Metamask -> Settings -> Advanced -> Clear activity tab data)
* Step 3. Navigate to Balance page
* Step 4. Click Add funds to the Colony and Copy Wallet address
* Step 5. Metamask -> send -> Paste Colony Wallet Address -> Send something to the colony (1 eth or if you choose xDai as Ganache Currency symbol choose xDai)
send-1-etch

> [!TIP]
> If you don't have ETH to send try to delete Account and Import it again
> If something doesn't work anyway, try to delete Ganache network and install it again

* Step 6. Navigate to Wayne Enterprises
* Step 7. Create simple payment from General to Colony Wallet Address that you copied at Step 4. 
image

* Step 8. Navigate back to Planet Express
* Step 9. Navigate to Incoming funds
* Step 10. Try to Accept this funds
* Step 11. Verify that after you click "Accept" or "Accept all available" all other buttons are disabled:
disabled-accept

> [!NOTE]
> I would recommend going through this flow twice: 1. Approve with the "Accept" button. 2. Approve with the "Accept all available" button. 
> I had another bug after pressing "Approve" in metamask before I started using values={getPayload} for the loading state. So, this is a weak place where something could go wrong.

Resolves #3313

### CODE REVIEW

File: src/components/frame/v5/pages/FundsPage/partials/AcceptButton/AcceptButton.tsx [hunk 1]
```diff
@@ -32,6 +36,11 @@ const AcceptButton: FC<AcceptButtonProps> = ({
     setIsClaimed(true);
     startPollingColonyData(1_000);
     setTimeout(stopPollingColonyData, 10_000);
+    disableAcceptLoading();
+  };
+
+  const getPayload = () => {
+    enableAcceptLoading();
   };
```
[the remark this answers is not in the document]
Comment (iamsamgibbs) [reply src/components/frame/v5/pages/FundsPage/partials/AcceptButton/AcceptButton.tsx:44]: Oh in fact, on further inspection, it does require a payload, and it comes from the transform function above. Maybe we could adapt this `getPayload` function to return those values and remove the transform function?

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Nice job fixing this up! I had a suspicion it would require a context when I was writing up the issue, so I'm glad that hunch was validated!

This works great:

https://github.com/user-attachments/assets/807dba38-5d01-4de8-a7bb-de5050c16a72

I think we could improve the accepting behaviour, the accept buttons become disabled before disappearing, but this is outside the scope of this issue:

https://github.com/user-attachments/assets/a8920454-da22-4d0b-ad96-a1bb173ff13a

I've left a minor comment regarding naming which I felt to be a bit unclear.

**Review by mmioana — CHANGES_REQUESTED** [review]
Awesome work on this one @Nortsova and thanks for picking this issue up 🥇 🫶 

Even though I didn't manage to properly test sending tokens from Metamask to the colony, I tried other 2 test scenarios

1. Creating a simple payment from Wayne to Planex
Screenshot 2024-10-28 at 13 25 08


https://github.com/user-attachments/assets/823cede5-67a2-4a13-a87f-77d184f0698e

2. Commenting the claim funds transactions from the `mint tokens` saga


Screenshot 2024-10-28 at 13 27 35

Screenshot 2024-10-28 at 13 28 02

Screenshot 2024-10-28 at 13 28 20

Screenshot 2024-10-28 at 13 28 25

https://github.com/user-attachments/assets/e58bbc5d-d486-441e-82a9-e84e0ff56394

Did notice, just as @iamsamgibbs, the `Accept` button gets re-enabled just before disappearing, but I guess this can be treated in a separate issue. 

Left a few minor refactoring comments, if you can please have a look 🙏 

Again really nice work 💯

File: src/components/frame/v5/pages/FundsPage/partials/AcceptButton/AcceptButton.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Nortsova) [reply src/components/frame/v5/pages/FundsPage/partials/AcceptButton/AcceptButton.tsx]: hey, thank you so much for your comment, I removed transform and just put values here, I will double-check tomorrow to not break anything. 🙌

**Comment by Nortsova** [comment]
Thank you guys for reviewing! I updated the code and created a ticket to disable the claiming button: 
https://github.com/JoinColony/colonyCDapp/issues/3502

**Review by iamsamgibbs — APPROVED** [review]
Thank you for the changes! This is still working great and the code reads more nicely now. 🎉

Screenshot 2024-10-29 at 10 56 01

**Review by mmioana — APPROVED** [review]
Thanks so much for your changes @Nortsova 🫶 really nice and clean work!

All accept buttons are disabled once an incoming funds entry is accepted

https://github.com/user-attachments/assets/2a2acb97-790a-47fe-be5f-56290e490343

**Review by jakubcolony — APPROVED** [review]
LGTM, nice work @Nortsova 💯 

As I understand the issue of being able to click on Accept straight after Accept all will be solved in #3502. I can confirm the buttons are disabled while transactions are in progress.

https://github.com/user-attachments/assets/f0a2a290-320a-4e83-93e7-3e3cac3c0342

**Pull request merged**